addons/point_of_sale/static/src/app/store/models.js

```diff
@@ -1297,6 +1363,19 @@ export class Order extends PosModel {
 
         if (options.json) {
             this.init_from_JSON(options.json);
+            for (const line of this.orderlines) {
+                line.comboLines = line?.combo_line_ids?.map((id) =>
+                    this.orderlines.find((l) => l?.id === id || l.cid === id)
+                );
+                const combo_parent_id = line.combo_parent_id?.length
+                    ? line.combo_parent_id[0]
+                    : line.combo_parent_id;
+                if (combo_parent_id) {
+                    line.comboParent = this.orderlines?.find(
+                        (l) => l?.id === combo_parent_id || l.cid === combo_parent_id
+                    );
+                }
+            }
```

Comment:
create a `lines_by_id` map and use it to find the corresponding line when populating the combo lines.